tests/testswitch.nim

```diff
@@ -625,42 +620,6 @@ suite "Switch":
     await allFuturesThrowing(
       switches.mapIt( it.stop() ))
 
-  # TODO: we should be able to test cancellation
-  # for most of the steps in the upgrade flow -
-  # this is just a basic test for dials
-  asyncTest "e2e canceling dial should not leak":
-    let ma = @[MultiAddress.init("/ip4/0.0.0.0/tcp/0").tryGet()]
-
-    let transport = TcpTransport.new(upgrade = Upgrade())
-    await transport.start(ma)
-
-    proc acceptHandler() {.async, gcsafe.} =
-      try:
-        let conn = await transport.accept()
-        discard await conn.readLp(100)
-        await conn.close()
-      except CatchableError:
-        discard
-
-    let handlerWait = acceptHandler()
-    let switch = newStandardSwitch(secureManagers = [SecureProtocol.Noise])
-
-    await switch.start()
-
-    var peerId = PeerId.init(PrivateKey.random(ECDSA, rng[]).get()).get()
-    let connectFut = switch.connect(peerId, transport.addrs)
-    await sleepAsync(500.millis)
```

Comment:
This test was not actually testing cancellation since the `connect` was finished far before 500 milliseconds